test flake: data race in TestFeaturesService_EntitlementsAPI

```
=== Failed
=== FAIL: enterprise/coderd TestFeaturesService_EntitlementsAPI/NoLicense (0.00s)
    testing.go:1319: race detected during execution of test
    --- FAIL: TestFeaturesService_EntitlementsAPI/NoLicense (0.00s)

=== FAIL: enterprise/coderd TestFeaturesService_EntitlementsAPI/FullLicense (0.00s)
    testing.go:1319: race detected during execution of test
    --- FAIL: TestFeaturesService_EntitlementsAPI/FullLicense (0.00s)

=== FAIL: enterprise/coderd TestFeaturesService_EntitlementsAPI/Warnings (0.00s)
    testing.go:1319: race detected during execution of test
    --- FAIL: TestFeaturesService_EntitlementsAPI/Warnings (0.00s)

=== FAIL: enterprise/coderd TestFeaturesService_EntitlementsAPI (0.00s)

=== FAIL: enterprise/coderd TestFeaturesServiceSyncEntitlements (0.00s)
    testing.go:1319: race detected during execution of test
==================
WARNING: DATA RACE
Write at 0x00c001483410 by goroutine 3840:
  github.com/coder/coder/enterprise/coderd.TestFeaturesService_EntitlementsAPI.func3()
      /home/coder/src/cdr/coder/enterprise/coderd/features_internal_test.go:139 +0x58d
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1446 +0x216
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1493 +0x47

Previous write at 0x00c001483410 by goroutine 3839:
  github.com/coder/coder/enterprise/coderd.TestFeaturesService_EntitlementsAPI.func2()
      /home/coder/src/cdr/coder/enterprise/coderd/features_internal_test.go:89 +0x509
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1446 +0x216
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1493 +0x47

Goroutine 3840 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1493 +0x75d
  github.com/coder/coder/enterprise/coderd.TestFeaturesService_EntitlementsAPI()
      /home/coder/src/cdr/coder/enterprise/coderd/features_internal_test.go:115 +0x81c
  testing.tRunner()
```

Both appear to refer to concurrent usage of `database.Store`.

--- a/enterprise/coderd/features_internal_test.go
+++ b/enterprise/coderd/features_internal_test.go
@@ -86,7 +86,7 @@ func TestFeaturesService_EntitlementsAPI(t *testing.T) {
 				auditLogs: entitlement{entitled},
 			},
 		}
-		_, err = db.InsertUser(ctx, database.InsertUserParams{
+		_, err := db.InsertUser(ctx, database.InsertUserParams{
 			ID:             uuid.UUID{},
 			Email:          "",
 			Username:       "",
@@ -136,7 +136,7 @@ func TestFeaturesService_EntitlementsAPI(t *testing.T) {
 			},
 		}
 		for i := byte(0); i < 5; i++ {
-			_, err = db.InsertUser(ctx, database.InsertUserParams{
+			_, err := db.InsertUser(ctx, database.InsertUserParams{
 				ID:             uuid.UUID{i},
 				Email:          "",
 				Username:       "",

